File: notes/17.feb/typescript/AssertionException.ts
export class AssertionException extends Error {

  private actual: Object;
  private expected: Object;

  public constructor(actual: Object, expected: Object, message: string) {
    super(message);
    this.actual = actual;
    this.expected = expected;
    this.message = `Message: ${this.message}\nActual: ${this.actual} Expected: ${this.expected}`
    this.name = "AssertionException";
  }

}

File: class exercises/17.feb/java/test/src/main/java/test/AssertionException.java
package test.src.main.java.test;

public class AssertionException extends RuntimeException {
  /**
   *
   */
  private static final long serialVersionUID = 1L;
  private Object actual;
  private Object expected;

  public AssertionException(String message, Object actual, Object expected) {
    super(message);
    this.actual = actual;
    this.expected = expected;
  }

  public Object getActual() {
    return actual;
  }

  public Object getExpected() {
    return expected;
  }
}